tests/server/test_middleware.py

```diff
@@ -60,3 +62,69 @@ def test_preflight_CORS_request(self):
                 list(response.headers.keys()),
                 msg=f"{response_header} header not found in response headers: {response.headers}",
             )
+
+
+class EnsureQueryParamIntegrityTest(SetClient, unittest.TestCase):
+
+    server = "regular"
+
+    def _check_error_response(
+        self,
+        request: str,
+        expected_status: int = None,
+        expected_title: str = None,
+        expected_detail: str = None,
+    ):
+        expected_status = 400 if expected_status is None else expected_status
+        expected_title = "Bad Request" if expected_title is None else expected_title
+        super()._check_error_response(
+            request, expected_status, expected_title, expected_detail
+        )
+
+    def test_wrong_html_form(self):
+        """Using a parameter without equality sign `=` or values should result in a `400 Bad Request` response"""
+        from optimade.server.query_params import EntryListingQueryParams
+
+        for valid_query_parameter in EntryListingQueryParams().__dict__:
+            request = f"/structures?{valid_query_parameter}"
+            with self.assertRaises(BadRequest):
+                self._check_error_response(
+                    request,
+                    expected_detail="A query parameter without an equal sign (=) is not supported by this server",
+                )
+
+    def test_wrong_html_form_one_wrong(self):
+        """Using a parameter without equality sign `=` or values should result in a `400 Bad Request` response
+
+        This should hold true, no matter the chosen (valid) parameter separator (either & or ;).
+        """
+        request = f"/structures?filter&include=;response_format=json"
+        with self.assertRaises(BadRequest):
+            self._check_error_response(
+                request,
+                expected_detail="A query parameter without an equal sign (=) is not supported by this server",
+            )
+
+    def test_parameter_separation(self):
+        """No matter the chosen (valid) parameter separator (either & or ;) the parameters should be split correctly"""
+        from optimade.server.middleware import EnsureQueryParamIntegrity
+
+        query_part = 'filter=""&include=;response_format=json'
```

Comment:
I get that this is a test for the middleware, but is this query valid?

`http://localhost:5000/optimade/v0/structures?filter=""&include=;response_format=json` gives me a 500 ParserError, as does `filter=''` i.e. single quotes.

Reply:
It is not - The `filter=""` does indeed give a `ParserError`, which is fine I guess? Or maybe it should actually be a `400 Bad Request`?
`response_format=json` should work though - weird.

Reply:
Testing this just now, `response_format=json` does indeed work. The `ParserError` comes from `filter`.
I have updated the string to be a working query string.

Reply:
Cheers, `response_format` worked for me too, looks like it unfortunately wrapped onto the next line in my comment